### ISSUE TITLE
Hyperparams in ImageNet experiment?

### ISSUE THREAD

**Issue by Jongchan** [google-research/uda#5]
Thank you for sharing the code!
I have several questions on hyper-parameter settings for ImageNet experiment.

1. What is **learning rate schedule, optimizer** for the ImageNet experiment? (including initial LR, total number of epochs or iterations, when to drop LR, and LR decay rate)
2. What is the **batch size (and LR)** for the experiments?
 2-1. Batch size for 10% supervised settings
 2-2. Batch size for semi-supervised settings (labeled set and unlabeled set)
3. **Loss balancing**? (coefficients for entropy loss, unsupervised loss and supervised loss)

**Comment by qizhex** [comment]
Sure!

1. learning_rate:
We follow standard practices for ResNet and decay the learning rate by 10 fold at 1/3, 2/3, 8/9 of the training process. We warmup the learning rate linearly until 5/90 of the training process. We use SGD Momentum for the optimizer.  We used this code [this function](https://github.com/tensorflow/tpu/blob/master/models/official/resnet/resnet_main.py#L211). 

2. Batch_size:
  10% setting: We use 512 for the supervised objective and 512 * 30 for the unsupervised objective. We set the Softmax temperature to 0.4 and set the confidence threshold to 0.5 (See the Additional training techniques in the paper for more details). The initial learning_rate is set to 0.3. We train for 40000 steps.
  Full data setting: We use 8192 for the supervised objective and 16384 for the unsupervised objective. For each unlabeled mini-batch,  1/4 of it is from ImageNet and 3/4 of it is from the 1.3M images of JFT. The initial learning rate is set to 1.6. We train for 42120 steps. 

3. Loss balancing:
  10% setting: the coefficient on supervised loss is 1. The coefficient on unsupervised loss is set to 20. We don't use entropy loss because of using the confidence threshold and Softmax temperature.
  Full data setting: the coefficient on supervised loss is 1. The coefficient on unsupervised loss is also set to 1. The coefficient for entropy minimization is set to 2 and is warmed up following the log-schedule defined in TSA section of the paper.
